lib/msf/core/payload/apk.rb

```diff
@@ -122,20 +147,20 @@ def backdoor_apk(apkfile, raw_payload)
       raise RuntimeError, "apktool version #{apk_v} not supported, please download at least version 2.0.1."
     end
 
-    unless File.readable?(File.expand_path("~/.android/debug.keystore"))
-      android_dir = File.expand_path("~/.android/")
-      unless File.directory?(android_dir)
-        FileUtils::mkdir_p android_dir
-      end
-      print_status "Creating android debug keystore...\n"
-      run_cmd("keytool -genkey -v -keystore ~/.android/debug.keystore \
-      -alias androiddebugkey -storepass android -keypass android -keyalg RSA \
-      -keysize 2048 -validity 10000 -dname 'CN=Android Debug,O=Android,C=US'")
-    end
-
     #Create temporary directory where work will be done
     tempdir = Dir.mktmpdir
 
+    keystore = "#{tempdir}/signing.keystore"
+    storepass = "android"
+    keypass = "android"
+    keyalias = "signing.key"
+    orig_cert_dname = find_orig_cert_dname(apkfile)
+
+    print_status "Creating signing key and keystore..\n"
+    run_cmd("keytool -genkey -v -keystore #{keystore} \
+    -alias #{keyalias} -storepass #{storepass} -keypass #{keypass} -keyalg RSA \
+    -keysize 2048 -validity 10000 -dname '#{orig_cert_dname}'")
```

Comment:
We could even can match the certificate start date and validity period. Something like:
```
orig_cert = find_orig_cert(apkfile)
dname = orig_cert[0]
startdate = orig_cert[1].strftime("%Y/%m/%d %H:%M:%S")
validity = (orig_cert[2] - orig_cert[1]) / (24 * 3600 * 1000) #?
run_cmd("keytool -genkey -v -startdate '#{startdate}
```
Annoyingly the dates are in different formats :(